core/src/main/scala/zio/config/ConfigDocsModule.scala

```diff
@@ -68,20 +85,14 @@ trait ConfigDocsModule extends WriteModule {
 
         docs match {
           case ConfigDocs.Leaf(sources, descriptions, _) =>
-            val desc =
-              descriptionsUsedAlready match {
-                case Some(value) =>
-                  descriptions.filter({
-                    case ConfigDocs.Description(path, _) if path.map(FieldName.Key) == Some(value) =>
-                      false
-                    case ConfigDocs.Description(_, _) => true
-                  })
-                case None => descriptions
-              }
+            val desc = filterDescriptions(descriptionsUsedAlready, descriptions)
 
             TableRow(previousPaths, Some(Table.Format.Primitive), desc, None, sources.map(_.name)).asTable
+          case ConfigDocs.Recursion(sources) =>
```

Comment:
My idea was that you either reach a `Leaf` or a `Recursion`.
If it would be `Recursive`, then (in my head) it would more like `Nested`, and I would expect it to not a leaf.